nextcord/types/snowflake.py

```diff
@@ -0,0 +1,22 @@
+# The MIT License (MIT)
+# Copyright (c) 2021-present vcokltfre & tag-epic
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from typing import Union
+
+Snowflake = Union[str, int]
```

Comment:
This should only be int and it should be a class so we can get things like created_at etc

Reply:
This should not only be int for consistency, since discord return IDs as strings: [See documentation here](https://discord.com/developers/docs/reference#snowflakes)

Reply:
doesnt the lib convert all to int for users

Reply:
The reason discord sometimes uses str for ints is that javascript doesnt support large ints that well, however python does so it makes no sense for us to store them in a less compact format. and honestly we shouldnt allow strs for ids.

Reply:
Let's go for int only, and a class (subclassing int??)